src/EbmlMaster.cpp

```diff
@@ -217,9 +217,6 @@ EbmlElement *EbmlMaster::FindFirstElt(const EbmlCallbacks & Callbacks, bool bCre
   if (bCreateIfNull) {
     // add the element
     EbmlElement *NewElt = &EBML_INFO_CREATE(Callbacks);
-    if (NewElt == nullptr)
```

Comment:
Yes but the Create callback can actually return *nullptr. Which hopefully is turned into nullptr when using &. IMO the proper fix would mean changing the API to return nullptr in the first place.

Reply:
Returning `*nullptr` is, if I'm not really mistaken, a violation of the specs, or at least undefined behavior. This API should really be changed to return one of:

* a regular pointer
* a smart pointer
* an `std::optional`

The difference between a pointer & a reference from the source code point of view is that references must always refer to an object, never to a `nullptr`. That they're both implemented via pointers under the hood is immaterial.

Reply:
Removed.